staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -251,16 +254,89 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
-			if len(rule.Message) != 0 {
-				errs = append(errs, field.Invalid(fldPath, sts.Type, rule.Message))
+			if compiled.MessageExpression != nil {
+				messageExpression, newRemainingBudget, msgErr := evalMessageExpression(ctx, compiled.MessageExpression, rule.MessageExpression, activation, remainingBudget)
+				if msgErr != nil {
+					if msgErr.errType == runtimeCostErr {
+						errs = append(errs, field.InternalError(fldPath, msgErr.err))
+						continue
+					} else {
+						klog.Warning(msgErr.err)
+						errs = append(errs, field.Invalid(fldPath, sts.Type, ruleMessageOrDefault(rule)))
```

Comment:
reuse `ruleErrorString`?

Reply:
I think this is also the same case as https://github.com/kubernetes/kubernetes/pull/115969#discussion_r1126809763.